cmd/cut-release/main.go

```diff
@@ -0,0 +1,131 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/plaid/go-envvar/envvar"
+)
+
+type envVars struct {
+	// Version is the new release version to use
+	Version string `envvar:"VERSION" default:"development"`
+}
+
+func main() {
+	env := envVars{}
+	if err := envvar.Parse(&env); err != nil {
+		log.Fatal(err)
+	}
+
+	updateHardCodedVersions(env.Version)
+
+	generateTypescriptClientDocs()
+}
+
+func generateTypescriptClientDocs() {
+	// Run `yarn install` to make sure `TypeDoc` dep is installed
+	cmd := exec.Command("yarn", "install")
+	cmd.Dir = "rpc/clients/typescript"
+	if err := cmd.Run(); err != nil {
+		log.Fatal(err)
+	}
+
+	// Run `yarn docs:md` to generate MD docs
+	cmd = exec.Command("yarn", "docs:md")
+	cmd.Dir = "rpc/clients/typescript"
+	if err := cmd.Run(); err != nil {
+		log.Fatal(err)
+	}
+
+	// Update SUMMARY.md
+	tsClientSummaryPath := "docs/json_rpc_clients/typescript/SUMMARY.md"
+	dat, err := ioutil.ReadFile(tsClientSummaryPath)
+	if err != nil {
+		log.Fatal(err)
+	}
+	// Modify the paths to be prefixed with `json_rpc_clients/typescript`
+	modifiedDat := strings.Replace(string(dat), "](", "](json_rpc_clients/typescript/", -1)
+	modifiedDat = strings.Replace(modifiedDat, "](json_rpc_clients/typescript/)", "]()", -1)
+	finalTsClientSummary := strings.Replace(modifiedDat, "* [", "  * [", -1)
+
+	// Replace the summary content nested under `Typescript client`
+	mainSummaryPath := "docs/SUMMARY.md"
+	dat, err = ioutil.ReadFile(mainSummaryPath)
```

Comment:
We could optionally use a [`buffio.Scanner`](https://golang.org/pkg/bufio/#Scanner) to cut down on memory usage. The current version loads the entire file into memory at once, but `buffio.Scanner` can load it into memory in pieces. Might not necessarily need to do it now, but It matters more if __SUMMARY.md__ is large.

```go
    summaryFile, err := os.Open("docs/SUMMARY.md")
    if err != nil {
        log.Fatal(err)
    }
    defer file.Close()

    scanner := bufio.NewScanner(summaryFile)
    for scanner.Scan() {
        // scanner.Text() tells you the current line.
    }
    if err := scanner.Err(); err != nil {
        log.Fatal(err)
    }
```

Of course, we would also need to figure out a way to replace/cut certain lines without loading the entire file into memory.

Reply:
This doesn't feel worthwhile for an utility command such as this. I also don't expect `SUMMARY.md` to ever be larger then twice it's current size (5kb).